lib/nimble_options/docs.ex

```diff
@@ -75,8 +76,46 @@ defmodule NimbleOptions.Docs do
   end
 
   defp get_default_str(schema) do
-    if Keyword.has_key?(schema, :default) do
-      "The default value is `#{inspect(schema[:default])}`."
+    if Keyword.has_key?(schema, :default),
+      do: "The default value is `#{inspect(schema[:default])}`."
+  end
+
+  defp get_type_str(schema) do
+    case schema[:type] do
+      nil ->
+        nil
+
+      {:custom, _module, _function, _args} ->
+        nil
+
+      {:fun, arity} ->
+        "function of arity #{arity}"
+
+      {:keyword_list, _} ->
+        "keyword list"
+
+      {:in, values} ->
+        "one of " <> Enum.map_join(values, ", ", &get_type_str(type: &1))
+
+      {:list, subtype} ->
+        if subtype_str = get_type_str(type: subtype) do
+          "list of #{subtype_str}"
+        end
+
+      {:non_empty_keyword_list, _} ->
+        "non-empty keyword list"
+
+      {:or, values} ->
+        Enum.map_join(values, " or ", &get_type_str(type: &1))
```

Comment:
This is problematic. Imagine I use `{:or, [:integer, {:custom, ...}]}`. The documentation is going to say `"integer or "`. Aside from the small formatting bug, the issue is that the documentation is now wrong, since it doesn't show potential valid values. Let's return `nil` here instead for simplicity.

```suggestion
        nil
```